### PR TITLE
fix: moved getAlertsAndRules inside of useAlerts hook

### PR DESCRIPTION
**Pull request by debjyoti-pandit**
fixes: https://github.com/red-hat-storage/odf-console/issues/368

### CODE REVIEW

**Comment by bipuladh** [comment]
To be merged after branch cut.
/hold

File: packages/shared/monitoring/useAlert.ts [hunk 1]
```diff
@@ -2,15 +2,55 @@ import {
...
+import * as _ from 'lodash';
+import { murmur3 } from 'murmurhash-js';
+
+type Group = {
+  rules: PrometheusRule[];
+  file: string;
+  name: string;
+};
+
+export type PrometheusRulesResponse = {
```
Comment (bipuladh) [inline packages/shared/monitoring/useAlert.ts:15]:
```suggestion
type PrometheusRulesResponse = {
```

File: locales/en/plugin__odf-console.json [hunk 2]
```diff
@@ -814,6 +814,8 @@
   "available": "available",
   "Other": "Other",
   "All other capacity usage that are not a part of the top 5 consumers.": "All other capacity usage that are not a part of the top 5 consumers.",
+  "For standalone MCG clusters, there is no set limit.": "For standalone MCG clusters, there is no set limit.",
+  "No Limit": "No Limit",
```
Comment (bipuladh) [inline locales/en/plugin__odf-console.json:818]: Where is this coming from?

**Review by bipuladh — CHANGES_REQUESTED** [review]

File: locales/en/plugin__odf-console.json (hunk 2 above)
Comment (SanjalKatiyar) [reply locales/en/plugin__odf-console.json:818]: got missed here: https://github.com/red-hat-storage/odf-console/pull/414
we can think about having some precommit check/CI build check for i18n as well...

Comment (debjyoti-pandit) [reply locales/en/plugin__odf-console.json]: It is gone, as I rebase it with the master branch. 
I agree with @SanjalKatiyar regarding adding a check in the CI as it shouldn't get missed.

File: packages/shared/monitoring/useAlert.ts [hunk 3]
```diff
@@ -2,15 +2,55 @@ import {
...
+  };
+  status: string;
+};
 
 const useAlerts = (basePath = '') => {
   const defaultBasePath = usePrometheusBasePath();
 
-  const [alerts, error, loading] = useCustomPrometheusPoll({
+  const [data, error, loading] = useCustomPrometheusPoll({
     endpoint: 'api/v1/rules' as any,
```
Comment (GowthamShanmugam) [inline packages/shared/monitoring/useAlert.ts:26]: replace 'api/v1/rules'  with PrometheusEndpoint.RULES

File: packages/shared/monitoring/useAlert.ts [hunk 4]
```diff
@@ -2,15 +2,59 @@ import {
...
-    query: 'api/v1/rules',
+  const [data, error, loading] = useCustomPrometheusPoll({
+    endpoint: PrometheusEndpoint.RULES as any,
+    query: PrometheusEndpoint.RULES,
     basePath: basePath || defaultBasePath,
   });
+
+  // Flatten the rules data to make it easier to work with, discard non-alerting rules since those
+  // are the only ones we will be using and add a unique ID to each rule.
+  const groups = (data as unknown as PrometheusRulesResponse)?.data?.groups;
```
Comment (GowthamShanmugam) [inline packages/shared/monitoring/useAlert.ts:37]: usePrometheusBasePath is always returning ProemetheusResponse type and converting to PrometheusRulesResponse type is not make much sense.

instead, convert the useCustomPrometheusPoll to receive the interface type.

```
import {
  DEFAULT_PROMETHEUS_SAMPLES,
  DEFAULT_PROMETHEUS_TIMESPAN,
  PrometheusEndpoint,
} from '@odf/shared/constants';
import { PrometheusResponse, PrometheusRule } from '@openshift-console/dynamic-plugin-sdk';
import { getPrometheusURL } from './helpers';
import { useURLPoll, } from './use-url-poll';

type CustomPrometheusResponse = PrometheusResponse | PrometheusRulesResponse;

type CustomPrometheusPollProps = {
  delay?: number;
  endpoint: PrometheusEndpoint;
  endTime?: number;
  namespace?: string;
  query: string;
  samples?: number;
  timeout?: string;
  timespan?: number;
  basePath?: string;
  cluster?: string;
};

type UseCustomPrometheusPoll = <P extends CustomPrometheusResponse>(
  props: CustomPrometheusPollProps
) => [P, any, boolean];

type Group = {
  rules: PrometheusRule[];
  file: string;
  name: string;
};

type PrometheusRulesResponse = {
  data: {
    groups: Group[];
  };
  status: string;
};


export const useCustomPrometheusPoll: UseCustomPrometheusPoll = <P extends CustomPrometheusResponse>({
  delay,
  endpoint,
  endTime,
  namespace,
  query,
  samples = DEFAULT_PROMETHEUS_SAMPLES,
  timeout,
  timespan = DEFAULT_PROMETHEUS_TIMESPAN,
  basePath,
  cluster,
}) => {
  const url = !!basePath
    ? getPrometheusURL(
        { endpoint, endTime, namespace, query, samples, timeout, timespan },
        basePath
      )
    : getPrometheusURL({
        endpoint,
        endTime,
        namespace,
        query,
        samples,
        timeout,
        timespan,
      });

  return useURLPoll<P>(url, delay, cluster, query, timespan);
};

```

Comment (GowthamShanmugam) [reply packages/shared/monitoring/useAlert.ts:37]: just invoking useCustomPrometheusPoll<PrometheusResonse> (or) useCustomPrometheusPoll<PrometheusRuleResonse> will takecare of the response type.

Comment (GowthamShanmugam) [reply packages/shared/monitoring/useAlert.ts:37]: @bipuladh Is this change too much or useful?

Comment (bipuladh) [reply packages/shared/monitoring/useAlert.ts:37]: I like this suggestion. Let's create a different issue for this.

**Review by bipuladh — APPROVED** [review]
/lgtm

**Pull request merged**